test/HowToWriteTestsUsingFileCheck.md

```diff
@@ -0,0 +1,100 @@
+# How to write tests using FileCheck
+
+## What is FileCheck
+
+FileCheck can be seen as an advanced version of grep. We use it for writing
+small annotated unit tests for optimization passes. FileCheck used in PyTorch is
+inspired by [LLVM FileCheck
+Tool](https://llvm.org/docs/CommandGuide/FileCheck.html), but is not the same.
+FileCheck is available for writing both C++ and python tests.
+
+## How does it work
+
+Let's look at a test written with FileCheck. The following test verifies that
+CSE pass removes one out of two similar `aten::mul` nodes. Here is how the test
+looks like:
+
+```python
+def test_cse():
+    input_str = """graph(%a : Tensor, %b : Tensor):
+      # CHECK: aten::mul
+      %x : Tensor = aten::mul(%a, %b)
+      # Check that the second aten::mul is removed by CSE.
+      # CHECK-NOT: aten::mul
+      %y : Tensor = aten::mul(%a, %b)
+      # CHECK: return
+      return (%x, %y)
+      """
+    parsed = parse_ir(input_str)
+    optimized = run_cse(parsed)
+    FileCheck().run(input_str, optimized)
```

Comment:
Could you add a comment about using the builder apI somewhere as well ? 
`FileCheck().check("aten::mul")`